### PR TITLE
Exercise 3

### PR DESCRIPTION
**Pull request by karlysun**

### CODE REVIEW

File: exercise-3.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+prompt.start();
+
+prompt.getAsync(['city']).then( function (result) {
+    
+        return result.city;
+        
+}).then( function(yourCity){
+
+    return request("https://maps.googleapis.com/maps/api/geocode/json?address="+yourCity);
```
Comment (ziad-saab) [inline exercise-3.js]: while this is OK, you can call `request` directly from the previous handler:

```javascript
return request("https://maps.googleapis.com/maps/api/geocode/json?address="+ ????);
```
I'll let you fill in the blanks :)

File: exercise-3.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    
+}).spread( function(res, body){
+    
+    var addressInfo = JSON.parse(body);
+    values[0] = addressInfo.results[0].geometry.location.lat;
+    values[1] = addressInfo.results[0].geometry.location.lng
+    console.log("You current location is: "+Math.floor(addressInfo.results[0].geometry.location.lat*100)/100+" x "+Math.floor(addressInfo.results[0].geometry.location.lng*100)/100);
+    return values;
```
Comment (ziad-saab) [inline exercise-3.js]: no need to return values if you're not using it. you can call the next handler's promise right here instead.

File: exercise-3.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    
+    return request("http://api.open-notify.org/iss-now.json");
+    
+}).spread( function(res, body) {
+    
+    var issInfo = JSON.parse(body);
+    values[2] = issInfo.iss_position.latitude;
+    values[3] = issInfo.iss_position.longitude;
+    console.log("The ISS is now at: "+Math.floor(issInfo.iss_position.latitude*100)/100+" x "+Math.floor(issInfo.iss_position.longitude*100)/100);
+    return values;
```
Comment (ziad-saab) [inline exercise-3.js]: no need to return values if not using it after

**Comment by ziad-saab** [comment]
@karlysun this is an OK solution but I think you can do better! The request for ISS position and the prompt could be done in parallel because they are independent. They can later be synchronized through a `Promise.join` like the previous exercise. I'd like you to revisit this at some point and make it better :)

**Comment by karlysun** [comment]
Updated!

**Comment by ziad-saab** [comment]
:+1:

**Pull request closed without merging**